### PR TITLE
Expose the `*redistore.RediStore`

### PR DESCRIPTION
**Pull request by SilverCory**
This PR will require the reverting of #52 via #54.

Here we provide the `*redistore.RediStore`, and I think this should be done in all sessions that use other libraries to allow more advanced functionality.

This works and has been tested breifly

### CODE REVIEW

File: redis/redis.go [hunk 1]
```diff
@@ -59,6 +60,29 @@ type store struct {
...
+	}
+
+	return
+}
+
+// SetKeyPrefix sets the key prefix in the redis database.
+func SetKeyPrefix(s Store, prefix string) error {
+	if err, rediStore := GetRedisStore(s); err != nil {
+		return err
+	} else {
```
Comment (appleboy) [inline redis/redis.go]: remove `else` block, just leave `rediStore.SetKeyPrefix(prefix)`

File: redis/redis.go [hunk 2]
```diff
@@ -59,6 +60,29 @@ type store struct {
...
+func GetRedisStore(s Store) (err error, rediStore *redistore.RediStore) {
+	if realStore, ok := s.(*store); !ok {
+		err = errors.New("unable to get the redis store: Store isn't *store")
+	} else {
+		rediStore = realStore.RediStore
+	}
+
+	return
+}
+
```
Comment (appleboy) [inline redis/redis.go]:
```
func GetRedisStore(s Store) (error, *redistore.RediStore) {
	realStore, ok := s.(*store)
	
	if !ok {
		return nil, errors.New("unable to get the redis store: Store isn't *store")
	}

	return realStore.RediStore, nil
}
```

File: redis/redis.go (hunk 1 above)
Comment (SilverCory) [reply redis/redis.go]: This won't work as you've done because rediStore is in the scope of the if statement, I will make a PR

**Comment by SilverCory** [comment]
I made those changes :)

**Pull request merged**